lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/LocalAccessControlModule.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.function.Supplier;
+
+import static io.trino.plugin.base.util.JsonUtils.parseJson;
+
+public class LocalAccessControlModule
+        implements Module
+{
+    @Override
+    public void configure(Binder binder) {}
+
+    @Inject
+    @Provides
+    @Singleton
+    public Supplier<AccessControlRules> getAccessControlRules(
+            FileBasedAccessControlConfig config)
+    {
+        File configFile = new File(config.getConfigFile());
```

Comment:
Do we verify if this `FileExists` as a part of the config ?

Reply:
No, as discussed we had to relax this validation as we have single shared config now. configFile might be local file or remote http endpoint now

Reply:
Not with annotations but as an additional validation.

Reply:
how exactly would you like to check it in config? As part of setter or getter?

Reply:
We can annotate a method with `AssertTrue` and check if the `File.exists` in that method (if it is a non-http file)

Reply:
done